### ISSUE TITLE
GetVmId:318: CreateLxProcess: WSL 2 requires an update to its kernel component.

### ISSUE THREAD

**Issue by boborjan2** [Biswa96/wslbridge2#39]
I get this error message while there is no update I could apply (at the moment). Also, my wsl2 setup runs without issues using the stock terminal.

c:\Users\x\AppData\Local\wsltty\bin>wslbridge2.exe
ERROR: GetVmId:318: CreateLxProcess: WSL 2 requires an update to its kernel component. For information please visit https://aka.ms/wsl2kernel

c:\Users\x\AppData\Local\wsltty\bin>wsl -v
WSL version: 1.1.2.0
Kernel version: 5.15.83.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.1265

**Comment by boborjan2** [comment]
I found out that there is one newer pre-release I could install. Same results:
c:\Users\x\AppData\Local\wsltty\bin>wsl -v
WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.1265

c:\Users\x\AppData\Local\wsltty\bin>wslbridge2.exe
ERROR: GetVmId:318: CreateLxProcess: WSL 2 requires an update to its kernel component. For information please visit https://aka.ms/wsl2kernel

**Comment by boborjan2** [comment]
And now I installed the latest stable release as well. Same issue:
c:\Users\x\AppData\Local\wsltty\bin>wsl -v
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.1265

c:\Users\x\AppData\Local\wsltty\bin>wslbridge2.exe
ERROR: GetVmId:318: CreateLxProcess: WSL 2 requires an update to its kernel component. For information please visit https://aka.ms/wsl2kernel

**Comment by Biswa96** [comment]
I can not reproduce the issue in my system running latest WSL 1.1.3.0.

> CreateLxProcess: WSL 2 requires an update to its kernel component.

The error message is not from wslbridge2. Do you have any extra settings in wsl.conf or in .wslconfig file? Also, what is the output of `wsl -l -v` command? You can hide any personal information with foo, bar like words.

**Comment by boborjan2** [comment]
C:\work\modules\TcpIp>wsl -l -v
  NAME          STATE           VERSION
* Debian        Running         2
  wsl-vpnkit    Stopped         2

.wslconfig:
[wsl2]
guiApplications=false
#kernelCommandLine = mitigations=off
#kernel=C:\\work\\linux\\vmlinux-5.10.102

wsl.conf: none

**Comment by Biswa96** [comment]
Nope, can't reproduce. Would you like to try the latest package from here https://github.com/microsoft/WSL/releases ? Uninstall the existing WSL before that. It'd not remove any Linux distributions in WSL, so generally safe to test.

Also, Which wslbridge2 version are you using and where did you download it from?

**Comment by boborjan2** [comment]
c:\Users\x\AppData\Local\wsltty\bin>wslbridge2.exe -h

wslbridge2 v0.10 : Runs a program within a Windows Subsystem for Linux (WSL) pty.
Copyright (C) 2019-2022 Biswapriyo Nath.
Licensed under GNU General Public License version 3 or later.

I use it as a part of wsltty. I does not start at all. I followed some issue descriptions and found that thi is where it fails.
I took version 1.1.3 from the WSL releases page. Seemingly it does not matter which version I use: 1.03, 1.1.2, 1.1.3 (see my comments regarding the versions I tried)

**Comment by Biswa96** [comment]
One quick workaround would be to install the msi package from that link https://aka.ms/wsl2kernel. It will install the Linux kernel in System32 directory. Generally it is not required with the new WSL from Microsoft Store.

**Comment by kaytabata** [comment]
Hello, I can reproduce this issue after applying KB5032190 with wsltty 3.7.0.
Uninstallation of the KB5032190 patch didn't solve this.

It seems that an undocumented COM API `CreateLxProcess()` fails.
I've reported this at https://github.com/mintty/wsltty/issues/327 as well.

I have time to help you with analysis and actual fix PR.
Can anyone give me some tips?
```
>wsl -l -v
  NAME                   STATE           VERSION
* Ubuntu                 Stopped         2
  docker-desktop-data    Stopped         2
  docker-desktop         Stopped         2
```
```
wsl -v
WSL version: 2.0.9.0
Kernel version: 5.15.133.1-1
WSLg version: 1.0.59
MSRDC version: 1.2.4677
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22631.2715
```

**Comment by Biswa96** [comment]
This has been fixed in latest v0.12 release https://github.com/Biswa96/wslbridge2/releases/tag/v0.12
